spyder/plugins/switcher/plugin.py

```diff
@@ -89,3 +146,85 @@ def on_main_menu_teardown(self):
 
     # --- Public API
     # ------------------------------------------------------------------------
+
+    # Switcher methods
+    def set_placeholder_text(self, text):
+        """Set the text appearing on the empty line edit."""
+        self._switcher.set_placeholder_text(text)
+
+    def setup(self):
+        """Set-up list widget content based on the filtering."""
+        self._switcher.setup()
+
+    def set_position(self, top, parent=None):
+        """Set the position of the dialog."""
+        self._switcher.set_position(top, parent)
+
+    # QDialog methods
+    def show(self):
+        """Show switcher widget with QT hide method for QDialog."""
+        self._switcher.show()
+
+    def hide(self):
+        """Hide switcher widget with QT hide method for QDialog."""
+        self._switcher.hide()
+
+    def close(self):
```

Comment:
Do we really need this method as part of the public API? I think with `hide` should be enough, right @dalthviz?

Reply:
Yes, I think one is enough, let me know if I should remove it.

Reply:
@dalthviz, what do you think about this?

Reply:
Sure, maybe even instead of overriding `close` here the container `on_close` method should be implemented doing this?

Reply:
okay, so should I remove the `hide()` method and leave only `on_close()` method? or should I keep both and just rename `close()` to `on_close()`?

Reply:
I think leaving the `hide` and renaming the close to `on_close` works :+1:

However, just in case, what do you think @ccordoba12 ?